src/App.jsx

```diff
@@ -14,45 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'aplphabet';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
+
+const buttons = [
+  {
+    title: 'Sort alphabetically',
+    value: SORT_FIELD_ALPHABET,
+    className: 'is-info',
+  },
+  {
+    title: 'Sort by length',
+    value: SORT_FIELD_LENGTH,
+    className: 'is-success',
+  },
+  {
+    title: 'Reverse',
+    value: SORT_FIELD_REVERSE,
+    className: 'is-warning',
+  },
+];
+
+const getPreparedGoods = (goods, sortField, isReversed) => {
+  const preparedGoods = [...goods];
+
+  switch (sortField) {
+    case SORT_FIELD_ALPHABET:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SORT_FIELD_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+      break;
+  }
+
+  return isReversed ? preparedGoods.reverse() : preparedGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {buttons.map(button => (
+          <button
+            key={button.value}
+            type="button"
+            className={cn('button', button.className, {
+              'is-light':
+                button.value === SORT_FIELD_REVERSE
+                  ? !isReversed
+                  : button.value !== sortField,
+            })}
+            onClick={() => {
+              if (button.value === SORT_FIELD_REVERSE) {
+                setIsReversed(!isReversed);
+              } else {
+                setIsReversed(false);
```

Comment:
Here is your problem, delete this string. Reverse must work untill you press it again.